File: Laba2/Task2.cs
using Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PZPK_Labas.Laba2
{
    public static class Task2
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 2. Задано масив. Знайти суму чисел, що знаходяться між максимальним і мінімальним елементами масиву \n" +
                "(в суму включити ці елементи). Вивести суму на екран.");

            int n;
            ConsoleInput.SafeIntInput("Введіть кількість елементів масиву", out n);

            int[] array = ArrayHelper.GenerateArray(n);

            Console.WriteLine("Масив: ");
            ArrayHelper.PrintArray(array);

            GetSumBetweenMaxAndMin(array);
        }

        private static void GetSumBetweenMaxAndMin(int[] array)
        {
            int maxIndex = 0;
            int minIndex = 0;
            int sum = 0;

            // Find max and min index
            for (int i = 0; i < array.Length; i++)
            {
                if (array[i] > array[maxIndex])
                    maxIndex = i;

                if (array[i] < array[minIndex])
                    minIndex = i;
            }

            Console.WriteLine("Максимальний елемент: " + array[maxIndex] + " | " + "Мінімальний елемент: " + array[minIndex]);


            int firstIndex = maxIndex > minIndex ? minIndex : maxIndex;
            int secondIndex = maxIndex < minIndex ? minIndex : maxIndex;

            for(int i = firstIndex; i <= secondIndex;i++)
                sum += array[i];

            Console.WriteLine("Cума між мінімальним і максимальним елементом: " + sum);
        }

    }
}

File: Laba2-Python/Task2.py
import ConsoleInput
import ArrayHelper
import sys

def Run():
    print("Завдання 2. Задано масив. Знайти суму чисел, що знаходяться між максимальним і мінімальним елементами масиву")
    print("(в суму включити ці елементи). Вивести суму на екран.")

    n = ConsoleInput.SafeIntInput("Введіть кількість елементів масиву")
    array = ArrayHelper.GenerateArray(n)

    print("Масив: ")
    ArrayHelper.PrintArray(array)

    get_sum_between_max_and_min(array)

def print_array(array):
    for i in range(len(array)):
        print(f"Елемент {i+1}: {array[i]}")

def get_sum_between_max_and_min(array):
    maxIndex = array.index(max(array))
    minIndex = array.index(min(array))
    sum = 0    

    firstIndex = min(maxIndex, minIndex)
    secondIndex = max(maxIndex, minIndex)

    for i in range(firstIndex, secondIndex+1):
        sum+= array[i]

    print("Cума між мінімальним і максимальним елементом:", sum)